internal/controller/pkg/revision/dependency.go

```diff
@@ -207,12 +209,16 @@ func (m *PackageDependencyManager) Resolve(ctx context.Context, pkg runtime.Obje
 			return found, installed, invalid, err
 		}
 		if !c.Check(v) {
-			invalidDeps = append(invalidDeps, lp.Identifier())
+			s := fmt.Sprintf("%s@%s", lp.Identifier(), lp.Version)
+			if dep.Constraints != "" {
+				s += " with " + strings.TrimSpace(dep.Constraints)
```

Comment:
Took me a bit to understand the notation, what about "expected ..." or desired/wanted maybe?

Reply:
Agreed. It would be ideal to state explicitly in the error:

* X version is currently installed.
* This package has a dependency on Y version.

Reply:
ping @sttts

Reply:
@phisco @negz I pushed a new commit (with @sttts permission) to clarify objects further in the message. Now it will look like:

```
❯ k get Configuration/jaredorg-config-xpkg -o json | jq '.status.conditions[] | select(.reason=="UnknownPackageRevisionHealth")'
{
  "lastTransitionTime": "2024-05-02T10:38:27Z",
  "message": "cannot resolve package dependencies: incompatible dependencies: existing package xpkg.upbound.io/crossplane-contrib/provider-kubernetes@v0.12.1 is incompatible with constraint >=v0.13.0",
  "reason": "UnknownPackageRevisionHealth",
  "status": "Unknown",
  "type": "Healthy"
}
```

What do you all think?